src/types.ts

```diff
@@ -1,5 +1,45 @@
-import { FunctionInput } from "@azure/functions";
+import { FunctionInput, FunctionHandler, FunctionOutput, FunctionTrigger } from "@azure/functions";
+import { IEntityFunctionContext } from "../src/ientityfunctioncontext";
+import { IOrchestrationFunctionContext } from "../src/iorchestrationfunctioncontext";
 
+// orchestrations
+export type OrchestrationHandler = (
+    context: IOrchestrationFunctionContext
+) => Generator<unknown, unknown, any>;
+
+export type OrchestrationOptions = {
+    handler: OrchestrationHandler;
+};
+
+export interface OrchestrationTrigger extends FunctionTrigger {
+    type: "orchestrationTrigger";
+}
+
+// entities
+export type EntityHandler<T> = (context: IEntityFunctionContext<T>) => void;
```

Comment:
Ideally entity shouldn't use generics either. Can we change it to `IEntityFunctionContext<T = unknown>` in `ientityfunctioncontext.ts` and remove the entity generics from this file? If not, I won't block on this since I know `IEntityFunctionContext` already had generics before your PR